test/unit/org/apache/cassandra/db/compaction/ValidationExecutorTest.java

```diff
@@ -79,7 +80,16 @@ public void testQueueOnValidationSubmission() throws InterruptedException
         while (threadsAvailable.get() > 0)
             TimeUnit.MILLISECONDS.sleep(10);
 
-        assertEquals(2, validationExecutor.getActiveTaskCount());
+        // getActiveTaskCount() relies on getActiveCount() which gives an approx number so we poll it
+        boolean activeTasksOk = false;
+        for (int i = 0; i < 100 && !activeTasksOk; i++)
+        {
+            TimeUnit.MILLISECONDS.sleep(10);
+            activeTasksOk = validationExecutor.getActiveTaskCount() == 2;
+        }
+        if (!activeTasksOk)
+            fail();
+
```

Comment:
The project as already an utility to perform that kind of check `Util.spinAssertEquals`. To see an example you can have a look into `ThreadPoolMetricsTest `

Reply:
+1 TIL